bookkeeper-server/src/main/java/org/apache/bookkeeper/discover/ZKRegistrationClient.java

```diff
@@ -253,17 +265,40 @@ public ZooKeeper getZk() {
 
     @SuppressWarnings("unchecked")
     private static BookieServiceInfo deserializeBookieService(byte[] bookieServiceInfo) {
-        if (bookieServiceInfo != null && bookieServiceInfo.length > 0) {
-            try {
-                return MAPPER.readValue(bookieServiceInfo, BookieServiceInfo.class);
-            } catch (IOException err) {
-                log.error("Cannot deserialize bookieServiceInfo from "
-                        + new String(bookieServiceInfo, StandardCharsets.UTF_8), err);
-            }
+        if (bookieServiceInfo == null || bookieServiceInfo.length == 0) {
+            // Empty BookieServiceInfoStructure
+            // ComponentInfoPublisher.EndpointInfo endpoint = new ComponentInfoPublisher.EndpointInfo("bookie", 
+            // localAddress.getPort(), localAddress.getHostName(), "bookie-rpc", null, extensions);
+            throw new RuntimeException("Not found");
         }
-        return BookieServiceInfo.EMPTY;
-    }
 
+        try {
+            BookieServiceInfoFormat.Builder builder = BookieServiceInfoFormat.newBuilder();
+            TextFormat.merge(new String(bookieServiceInfo, UTF_8), builder);
```

Comment:
Not sure this is the better way.
Check on other parts,
we are supporting both binary and text encoding for protobuf
writing as binary it is actually better, because it supports schema evolution